### DISCUSSION TITLE
All fonts track Linux Libertine serifs

### DISCUSSION THREAD

**Discussion by context-dependent** [typst/typst#3761] [General]
### Description

In the reference image below, the left edge of each text block using default font (Linux Libertine) lines up exactly with the left edge of its container.
The text blocks using other fonts don't, instead leaving room for LL's large serifs. 
This is most obvious in the two sans fonts tested (Arial and Segoe UI), but it appears to be true of other serif fonts as well.
The misalignment of the left edge is clear, but it feels like it could be affecting the tracking throughout the text. 

Apologies if this is a known issue. I couldn't find it in the repo, but I don't have the font design terminology to be confident it hasn't already been flagged. 

## Reference Image

![image](https://github.com/typst/typst/assets/31548710/d69abdf1-2a65-4c08-9911-c19f3894aae4)

## Typst code

```typst
Default serif

#box(fill: rgb("#cff7ff"), width: 100%)[
  #block(text(size: 50pt)[III (50pt)])
  #block(text(size: 28pt)[III (28pt)])
  #block(text(size: 16pt)[III (16pt)])
  #block(text(size: 11pt)[III (11pt)])  
]

#v(20pt)


#text(font: "Garamond")[Garamond]

#box(fill: rgb("#cff7ff"), width: 100%)[
  #block(text(font: "Garamond", size: 50pt)[III (50pt)])
  #block(text(font: "Garamond", size: 28pt)[III (28pt)])
  #block(text(font: "Garamond", size: 16pt)[III (16pt)])
  #block(text(font: "Garamond", size: 11pt)[III (11pt)])  
]

#v(20pt)


#text(font: "Arial")[Arial]

#box(fill: rgb("#cff7ff"), width: 100%)[
  #block(text(font: "Arial", size: 50pt)[III (50pt)])
  #block(text(font: "Arial", size: 28pt)[III (28pt)])
  #block(text(font: "Arial", size: 16pt)[III (16pt)])
  #block(text(font: "Arial", size: 11pt)[III (11pt)])  
]

#v(20pt)

#text(font: "Segoe UI")[Segoe UI]

#box(fill: rgb("#cff7ff"), width: 100%)[
  #block(text(font: "Segoe UI", size: 50pt)[III (50pt)])
  #block(text(font: "Segoe UI", size: 28pt)[III (28pt)])
  #block(text(font: "Segoe UI", size: 16pt)[III (16pt)])
  #block(text(font: "Segoe UI", size: 11pt)[III (11pt)])  
]
```

### Operating system

Windows

### Typst version

- [X] I am using the latest version of Typst

**Comment by EpicEricEE** [comment]
I think it's rather a quirk of the sans-serif fonts themselves, as it looks the same in plain HTML (with Arial):

![image](https://github.com/typst/typst/assets/7191192/616626ad-f038-4204-8789-bbfbc5170a9b)

**Comment by context-dependent** [comment]
Good to know! 

In case its useful to anyone, I found a workaround that adds a negative left inset to the heading block. 

![image](https://github.com/typst/typst/assets/31548710/a669f1f7-5386-4348-84fe-54ac3087797a)

```typst
#let sans-heading(x, font: "Arial", size: 11pt) = {
  let i = (size / -11)
  block(
    inset: (left: i), 
    text(
      font: font, 
      size: size, 
      x
    )
  )
}

#box(fill: rgb("#cff7ff"), width: 100%)[
  #for size in (11pt, 16pt, 28pt, 50pt) {
    sans-heading(size: size)[III (#size)]
  }
]
```

**Reply by emilyyyylime** [reply]
You can also use `context` to improve this such that explicitly passing the styles won't be necessary

**Reply by context-dependent** [reply]
Not explicitly passing the styles seems great! Could you share an example of how you might do that?
